Source/Core/Core/PowerPC/BreakPoints.cpp

```diff
@@ -120,19 +120,32 @@ void BreakPoints::Add(u32 address, bool temp)
 void BreakPoints::Add(u32 address, bool temp, bool break_on_hit, bool log_on_hit,
                       std::optional<Expression> condition)
 {
-  // Only add new addresses
-  if (IsAddressBreakPoint(address))
-    return;
-
-  TBreakPoint bp;  // breakpoint settings
-  bp.is_enabled = true;
-  bp.is_temporary = temp;
-  bp.break_on_hit = break_on_hit;
-  bp.log_on_hit = log_on_hit;
-  bp.address = address;
-  bp.condition = std::move(condition);
+  // Check for existing breakpoint, and overwrite with new info.
+  // This is assuming we usually want the new breakpoint over an old one.
+  auto iter = std::find_if(m_breakpoints.begin(), m_breakpoints.end(), [address](const auto& bp) {
+    return bp.address == address;
+  });
 
-  m_breakpoints.emplace_back(std::move(bp));
+  if (iter != m_breakpoints.end())  // We found an existing breakpoint
+  {
+    TBreakPoint &bp = *iter;  // breakpoint settings
+    bp.is_temporary = temp;
+    bp.break_on_hit = break_on_hit;
+    bp.log_on_hit = log_on_hit;
+    bp.address = address;
+    bp.condition = std::move(condition);
```

Comment:
Making the temp bp then doing `*iter = std::move(bp);` would probably work better than writing the assignments twice

Reply:
What about the is_enabled flag? Currently I have it not be affected when applying an edit, but doing an std::move into the breakpoint would force it to be re-enabled.

Reply:
Nvm I just pulled the flag before moving